soc/arm/microchip_mec/common/spigen/mec_spi_gen.py

```diff
@@ -0,0 +1,498 @@
+
+    if VERBOSE_MODE:
+        print_bytes("hdr_sha384_digest", hdr_digest)
+
+    hdr[0xB0:0xE0] = hdr_digest
+
+    return bytes(hdr)
+# end build_header
+
+def main():
+    """MEC SPI Gen"""
+
+    parser = argparse.ArgumentParser()
+    # Use a lambda to handle base 10 or base 16 (hex) input
+    parser.add_argument("-c",
+                        type=str,
+                        dest="chip",
+                        default="mec172x",
+                        help="Chip name: mec172x(default) or mec152x")
+    parser.add_argument("-i",
+                        type=str,
+                        dest="infilename",
+                        default="zephyr.bin",
+                        help="Input firmware binary file path/name. Default=zephyr.bin")
+    parser.add_argument("-o",
+                        type=str,
+                        dest="outfilename",
+                        default="zephyr.mchp.bin",
+                        help="Output SPI image file path/name. Default=zephyr.mchp.bin")
+    parser.add_argument("-s",
+                        type=int,
+                        dest="spi_size_kb",
+                        default=256,
+                        help="SPI image size in kilobytes. Default=256")
+    parser.add_argument("-e",
+                        type=int,
+                        dest="entry_point",
+                        default=0,
+                        help="FW entry point address. Default=0, Lookup in image")
+    parser.add_argument("-f",
+                        type=int,
+                        dest="spi_freq_mhz",
+                        default=12,
+                        help="SPI frequency: 12, 16, 24, or 48 MHz")
+    parser.add_argument("-r",
+                        type=str,
+                        dest="spi_read_mode",
+                        default="fast",
+                        help="SPI read mode: normal, fast, dual or quad")
+    parser.add_argument("-m",
+                        type=int,
+                        dest="spi_mode",
+                        default=0,
+                        help="SPI signalling mode 3-bit field: 0-7")
+    parser.add_argument("--drvstr",
+                        type=str,
+                        dest="spi_drive_strength",
+                        default="1x",
+                        help="SPI pin driver strength multiplier encoded")
+    parser.add_argument("--slewrate",
+                        type=str,
+                        dest="spi_slew_rate",
+                        default="slow",
+                        help="SPI pins slew rate")
+    parser.add_argument("--fill",
+                        dest="fill",
+                        action='store_true',
+                        help="Fill with 0xFF to flash size")
+    parser.add_argument("-v",
+                        dest="verbose",
+                        action='store_true',
+                        help="Enable messages to console")
+
+    args = parser.parse_args()
```

Comment:
move this block to a separate function? (most other scripts do that)

Reply:
ok

Reply:
added separate function